libmachine/persist/secretstore.go

```diff
@@ -0,0 +1,210 @@
+package persist
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rancher/machine/libmachine/host"
+	"github.com/rancher/machine/libmachine/log"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1types "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type secretStore struct {
+	Store
+	SecretName, SecretNamespace string
+	SecretClient                corev1types.SecretInterface
+	secret                      *v1.Secret
+}
+
+func NewSecretStore(store Store, secretName, secretNamespace, kubeConfigPath string) (Store, error) {
+	var config *rest.Config
+	var err error
+	if kubeConfigPath != "" {
```

Comment:
BuildConfigFromFlags checks if kubeconfigpath is empty and returns the InClusterConfig if empty, can probably toss this if/else check and just check